### DISCUSSION TITLE
Are multiple AppShell's a good option/practice?

### DISCUSSION THREAD

**Discussion by DWhiteR1** [dotnet/maui#16481] [Q&A]
In my app I have 3 Shells StartUpAppShell, UnAutherizedAppShell and AppShell
1. StartUpAppShell is just a Place holder Page "Loading Application Essentials..." Which Checks if the user is authenticated and a few other things.
If the user is not authenticated then.
2. UnAutherizedAppShell has two ShellContent's one for Login and another for Register.
3. AppShell has all of my application pages.
I use DI to create UnAutherizedAppShell and AppShell
        
`builder.Services.AddSingleton<AppShell>();`
`builder.Services.AddSingleton<UnAutherizedAppShell>();`

Then in my StartUpViewModel I retrieve and assign the relevant shell as required.
`AppShell appShell = Application.Current.Handler.MauiContext.Services.GetService<AppShell>();`

`Application.Current.MainPage = appShell();`

This works well for me at the moment, but its early days and I don't want to stray too far down a (Not Recomended Path.)

Just putting it out there if anyone would like to comment.

Regards,
Dean

**Comment by dgerding** [comment]
I've been working for a while with two, one desktop and one shell, but as of a week ago I found a combination that works well on both with just platform based styling. I'm going to keep the two different classes in case I decide to add "desktop or mobile" specific shell tweaks.

that's my experience. If the question is, does it cause trouble: not really. More work. Yep.

Dave

**Reply by DWhiteR1** [reply]
Thanks for your Input Dave.
I do have one Issue but I dont know if its related to multiple appShell.
If I log out, Which does the switch and log back in again.
My Title bar on WinUI goes missing.
I'm still investigating, so we will see.
If I find the issue, I'll use this thread to update.
I'll leave this unanswered for a few days to see if there is any further input.
Thanks for taking the time to comment.
Regards,
Dean.
